### PR TITLE
Added TriMakerV2 + Test

### PR DESCRIPTION
**Pull request by 0xdon**
**TriMakerV2 - Converts LP fees into USDC**
- Sends USDC to StableTriMaker (to be specified in constructor on deploy)
- Has a reclaim functionality for owner to withdraw tokens in case of emergency
- Owner can change recipient stabletrimaker address

### CODE REVIEW

**Comment by rej156** [comment]
I know it's a pain but mind changing the test cases to swap into USDC even though TRI uses the same erc20mock contract?

just rename most of the vars tbh

just to reflect intended functionality imo

File: scripts/deployTriMakerV2.ts
Comment (0xchain1) [inline scripts/deployTriMakerV2.ts:28]: put this in constants

File: scripts/deployTriMakerV2.ts
Comment (0xchain1) [inline scripts/deployTriMakerV2.ts:34]: delete if not used

File: contracts/rewards/TriMakerV2.sol
Comment (0xchain1) [inline contracts/rewards/TriMakerV2.sol:21]: stableUsnMaker 
- change this everywhere -- scripts, test, contract

File: scripts/constants.ts
Comment (baboobhaiya) [inline scripts/constants.ts:12]: what is the process here? should we wait for this PR to be merged before `stableUsnMaker` gets merged+launched or lmerge this PR?

**Comment by baboobhaiya** [comment]
Can we rename `TriMaker` to UsdcMaker` ?

File: contracts/rewards/TriMakerV2.sol
Comment (baboobhaiya) [inline contracts/rewards/TriMakerV2.sol:84]: lets remove this, since we can always change the `FeeTo` address in UniswapFaktory 
if we want to change StableUsnMaker, lets deploy a new contract and take it from there

Comment (baboobhaiya) [reply contracts/rewards/TriMakerV2.sol:84]: trying to reduce the attack surface

File: scripts/constants.ts
Comment (0xchain1) [reply scripts/constants.ts:12]: ya, we can merge this and then add another PR?

**Review by baboobhaiya — APPROVED** [review]

**Review by rej156 — APPROVED** [review]

**Pull request merged**
